mathics/builtin/comparison.py

```diff
@@ -219,75 +219,65 @@ def numerify_args(items, evaluation):
-        # [https://github.com/mathics/Mathics/pull/1209#issuecomment-810277502]
-        # for a future refactory of this methods...
-        if hasattr(lhs, "equal2"):
-            result = lhs.equal2(rhs)
-            if result is not None:
-                return result
-
-        # Do this after lhs.equal2 since lhs.equal2 will do a sameQ test.
-        if lhs.sameQ(rhs):
-            return True
-
-        # FIXME: I think this can be folded into Symbol.equal2 and Atom.equal2
-        if not (isinstance(lhs, Symbol) or isinstance(rhs, Symbol)) and (
-            isinstance(lhs, Atom) and isinstance(rhs, Atom)
-        ):
-            return lhs == rhs
-
-        # Dealing with two non-atomic expressions:
-        if not rhs.is_atom():
-            head1 = lhs.get_head()
-            head2 = rhs.get_head()
-
-            # FIXME: add equal2 method for DirectedInfinity
-
-            # Handling comparisons with DirectedInfinity
-            if head2.sameQ(SymbolDirectedInfinity):
-                lhs, rhs = rhs, lhs
-                head1, head2 = head2, head1
-            if head1.sameQ(SymbolDirectedInfinity):
-                if head2.sameQ(SymbolDirectedInfinity):
-                    dir1 = dir2 = Integer(1)
-                    if len(lhs._leaves) == 0:
-                        if len(rhs._leaves) == 0:
-                            return True
-                        dir1 = Integer(1)
-                    else:
-                        dir1 = lhs._leaves[0]
-                    if len(rhs._leaves) == 0:
-                        if dir1.sameQ(Integer(1)):
-                            return True
-                        dir2 = Integer(1)
-                    else:
-                        dir2 = rhs._leaves[0]
-                    # If the directions are equal,
-                    # then both infinites are the same
-                    if self.equal2(dir1, dir2):
-                        return True
-                    # Now, compare the signs:
-                    dir1 = Expression("Sign", dir1)
-                    dir2 = Expression("Sign", dir2)
-                    return self.equal2(dir1, dir2)
-
-        # Dealing with one expression
-        elif not lhs.is_atom():
-            # FIXME: Ass mentioned above: add equal2 method for DirectedInfinity
-            if lhs.get_head().sameQ(SymbolDirectedInfinity):
-                if isinstance(rhs, Number):
-                    return False
-                elif SymbolInfinity.sameQ(rhs):
-                    if len(lhs._leaves) == 0 or self.equal2(lhs._leaves[0], Integer(1)):
-                        return True
-
-        lhs_sympy = lhs.to_sympy(evaluate=True, prec=COMPARE_PREC)
-        rhs_sympy = rhs.to_sympy(evaluate=True, prec=COMPARE_PREC)
+    @staticmethod
+    def get_pairs(args):
+        for i in range(len(args)):
+            for j in range(i):
+                yield (args[i], args[j])
+
+    def expr_eq_tst(self, lhs, rhs, max_extra_prec=None) -> Optional[bool]:
```

Comment:
I am sorry I wasn't clearer here we want `expr_equal_txt` here and not `eq` .

In Lisp and symbolic languages which `eq` is `WL SameQ`. But we ahve been using sameQ inside the Python code already. 

If inside some `..eq()` function we are calling `equal2` that is definitiely a clue that we mean `equal` not `eq`. And if we are only call `sameQ` or do sameQ kinds of tests, the the word `sameQ` would be used. 

Either way, don't use `eq` which we happen not to be using. I was just pointing out that this is what other languages typically use for WL `SameQ`.

Reply:
OK, so ``_equal``  instead of ``_eq_tst`` would be OK?

Reply:
`_equal` everyplace `_eq_test` is used would be awesome!

Sorry to put you through all of this trouble. (I think though it is all for the better and worth it in the long run.)